packages/queued-request-controller/tests/QueuedRequestController.test.ts

```diff
@@ -0,0 +1,224 @@
+        }
+      });
+
+      it('handles errors without interrupting the execution of the next item in the queue', async () => {
+        const options: QueuedRequestControllerOptions = {
+          messenger: buildQueuedRequestControllerMessenger(),
+        };
+
+        const controller = new QueuedRequestController(options);
+
+        // Mock requests with one request throwing an error
+        const request1 = jest.fn(async () => {
+          await new Promise((resolve) => setTimeout(resolve, 100));
+        });
+
+        const request2 = jest.fn(async () => {
+          throw new Error('Request 2 failed');
+        });
+
+        const request3 = jest.fn(async () => {
+          await new Promise((resolve) => setTimeout(resolve, 50));
+        });
+
+        // Enqueue the requests
+        const promise1 = controller.enqueueRequest(request1);
+        const promise2 = controller.enqueueRequest(request2);
+        const promise3 = controller.enqueueRequest(request3);
+
+        try {
+          // Wait for all promises to resolve (expect the second one to throw an error)
+          await Promise.all([promise1, promise2, promise3]);
+        } catch (error: any) {
+          // Assert that the error is from request2
+          expect(error.message).toBe('Request 2 failed');
+        }
+
+        // Ensure that request3 still executed despite the error in request2
+        expect(request1).toHaveBeenCalled();
+        expect(request2).toHaveBeenCalled();
+        expect(request3).toHaveBeenCalled();
+      });
+    });
+  });
+
+  describe('length', () => {
+    it('emits countChanged event when the queue length changes', async () => {
+      const options: QueuedRequestControllerOptions = {
+        messenger: buildQueuedRequestControllerMessenger(),
+      };
+
+      const controller = new QueuedRequestController(options);
+
+      // Mock the event listener
+      const eventListener = jest.fn();
+
+      // Subscribe to the countChanged event
+      options.messenger.subscribe(
+        'QueuedRequestController:countChanged',
+        eventListener,
+      );
+
+      // Enqueue a request, which should increase the count
+      controller.enqueueRequest(
+        async () => new Promise((resolve) => setTimeout(resolve, 10)),
+      );
+
+      // Check if the event was emitted with the correct parameters (increased count)
+      expect(eventListener).toHaveBeenCalledWith(1);
+
+      // Enqueue another request, which should increase the count
+      controller.enqueueRequest(
+        async () => new Promise((resolve) => setTimeout(resolve, 10)),
+      );
+
+      // Check if the event was emitted again with the correct parameters (increased count)
+      expect(eventListener).toHaveBeenCalledWith(2);
+
+      // Resolve the first request, which should decrease the count
+      await new Promise((resolve) => setTimeout(resolve, 10)); // Simulate request completion
+      expect(eventListener).toHaveBeenCalledWith(1);
```

Comment:
call order matters for these expects. For example, this expect will still pass because it was called with 1 on line 148 regardless of what the side effect of line 164 is

Reply:
yeah thats a good bit of detail to add Ill give it a try